### PR TITLE
feat: create `trunk` completion spec

### PR DESCRIPTION
**Pull request by t-bre**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
Feature

**What is the current behavior? (You can also link to an open issue here)**
No completion spec for `trunk`

**What is the new behavior (if this is a feature change)?**
Creates completion spec for `trunk`

**Additional info:**
[trunk website](https://trunk.io/products/check)

### CODE REVIEW

File: src/trunk.ts [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+      name: "--no-progress",
+      description: "Don't show progress updates",
+      isPersistent: true,
+    },
+    {
+      name: "--ci-progress",
+      description:
+        "Show updates every 30 seconds without clearing terminal screen (implicit with --ci)",
+      isPersistent: true,
+    },
```
Comment (fedeci) [inline src/trunk.ts:274]: Are these mutually [exclusiveOn](https://fig.io/docs/reference/option#exclusiveon)

File: src/trunk.ts [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+          description: "Add sha256s to trunk.yaml for additional verification",
+        },
+        {
+          name: "--check-sample",
+          description: "Run `trunk check sample` without prompting",
+        },
+        {
+          name: "--nocheck-sample",
+          description: "Do not run `trunk check sample` post-init",
+        },
```
Comment (fedeci) [inline src/trunk.ts:96]: These too are likely `exclusiveOn`

**Review by fedeci — CHANGES_REQUESTED** [review]
Thanks for this spec, I left two comments to be solved before merging but overall it looks good!

File: src/trunk.ts (hunk 1 above)
Comment (t-bre) [reply src/trunk.ts]: Yes you're right, fixed

File: src/trunk.ts (hunk 2 above)
Comment (t-bre) [reply src/trunk.ts]: Same here, also fixed

File: src/trunk.ts [hunk 3]
```diff
@@ -0,0 +1,295 @@
...
+      "Upstream branch used to compute changed files (autodetected by default)",
+    args: {
+      name: "branch",
+    },
+  },
+];
+
+const completionSpec: Fig.Spec = {
+  name: "trunk",
+  description: "",
```
Comment (SeparateRecords) [inline src/trunk.ts:79]: Would you be able to add a description here?

File: src/trunk.ts [hunk 4]
```diff
@@ -0,0 +1,295 @@
...
+      isPersistent: true,
+    },
+    {
+      name: ["-m", "--monitor"],
+      description:
+        "Enable the trunk daemon to monitor file changes in your repo",
+      isPersistent: true,
+      args: {
+        name: "value",
+        suggestions: ["0", "1"],
```
Comment (SeparateRecords) [inline src/trunk.ts:249]: Should this be `true`/`false` instead of `0`/`1`?

File: src/trunk.ts [hunk 5]
```diff
@@ -0,0 +1,295 @@
...
+    },
+    {
+      name: "--ci-progress",
+      description:
+        "Show updates every 30 seconds without clearing terminal screen (implicit with --ci)",
+      isPersistent: true,
+      exclusiveOn: ["--no-progress"],
+    },
+    {
+      name: "--action_timeout",
```
Comment (SeparateRecords) [inline src/trunk.ts:280]: All the other options use dashes, should this be a dash instead of underscore?

File: src/trunk.ts [hunk 6]
```diff
@@ -0,0 +1,295 @@
...
+  options: [
+    {
+      name: ["-h", "--help"],
+      description: "Usage information",
+      isPersistent: true,
+    },
+    {
+      name: "--version",
+      description: "The version",
+      isPersistent: true,
```
Comment (SeparateRecords) [inline src/trunk.ts:240]: Is this persistent on each subcommand? Usually `--version` is only on the root (I don't have trunk installed to check)

**Review by SeparateRecords — CHANGES_REQUESTED** [review]
Looks good! Couple of quetions in reviews, but aside from that this spec is in a really good place :)

File: src/trunk.ts (hunk 4 above)
Comment (t-bre) [reply src/trunk.ts]: Good point, just checked and both ways work. I think `true`/`false` is more intuitive, so I've changed it to that.

File: src/trunk.ts (hunk 3 above)
Comment (t-bre) [reply src/trunk.ts]: Yes sorry I missed that, done now.

File: src/trunk.ts (hunk 5 above)
Comment (t-bre) [reply src/trunk.ts:281]: Yeah I thought that as well, but no for some reason this one has an underscore. Tried it with a dash and it throws an error. According to the help text `--action_timeout` is correct:

```
usage
  trunk [flags] [subcommand]

flags
  -h, --help                  usage information
      --version               the version
  -m, --monitor               enable the trunk daemon to monitor file changes in your repo
      --ci                    run in continuous integration mode
  -o, --output                output format {text | json}
      --no-progress           don't show progress updates
      --ci-progress           show updates every 30 seconds without clearing terminal screen (implicit with --ci
      --action_timeout        how long actions (downloads, lint runs, etc.) have to time out
  -v, --verbose               output details about what's happening under the hood
```

Comment (SeparateRecords) [reply src/trunk.ts:281]: That's super weird! Thanks for double checking 😄

File: src/trunk.ts (hunk 6 above)
Comment (SeparateRecords) [reply src/trunk.ts:241]: Neat! Unusual behavior but good to have it modeled perfectly.

**Review by SeparateRecords — COMMENTED** [review]
LGTM! Thanks so much for contributing this 😄

**Review by SeparateRecords — APPROVED** [review]

**Pull request merged**